pyfgaws/batch/api.py

```diff
@@ -340,13 +344,17 @@ def wait_on(
         self,
         status_to_state: Dict[Status, bool],
         max_attempts: Optional[int] = None,
-        delay: Optional[int] = None,
+        delay: Optional[Union[int, float]] = None,
         after_success: bool = False,
     ) -> batch.type_defs.JobDetailTypeDef:
         """Waits for the given states with associated success or failure.
 
         If some states are missing from the input mapping, then all statuses after the last
-        successful input status are treated as success or failure based on `after_success`
+        successful input status are treated as success or failure based on `after_success`.
+
+        This method adds a small random jitter (+/-2 seconds) to the delay, enforcing a minimum
+        delay of 1 second, to help avoid AWS batch API limits for monitoring batch jobs in the
+        cases of many requests across concurrent jobs.
```

Comment:
Maybe worth noting in the `delay` help text that a delay of 0 will be set to `MINIMUM_DELAY`, if my change is accepted, or fail, if it is not.